projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts

```diff
@@ -1,12 +1,134 @@
   selector: 'lib-certificates',
   standalone: true,
-  imports: [],
+  imports: [TranslateModule, MatIconModule, MatRadioModule, MatSelectModule, MatFormFieldModule, FormsModule, ReactiveFormsModule, CommonModule, MatDialogModule, MatInputModule],
   templateUrl: './certificates.component.html',
-  styleUrl: './certificates.component.css'
+  styleUrl: './certificates.component.scss'
 })
-export class CertificatesComponent {
+export class CertificatesComponent implements OnInit{
+  certificateDetails : any;
+  certificateForm !: FormGroup 
+  attachLogo: Array<{ name: string }> = [];
+  attachSign: Array<{ name: string }> = [];
+  certificateTypeSelected = false;
+  filespecifications = [
+    { no: '1', option: "HEIGHT_SPECIFICATION" },
+    { no: '2', option: "SIZE_SPECIFICATIION" },
+    { no: '3', option: "FILE_SPECIFICATION" } 
+  ]
+  evidenceNumber = [1, 2, 3];
+
+  constructor(private dialog : MatDialog, private fb: FormBuilder, private formService: FormService,){}
+
+  ngOnInit() {
+    this.initForm();
+    this.getCertificateForm();
+  }
+
+  initForm() {
+    this.certificateForm = this.fb.group({
+      selectedOption: [''],
+      certificateType: ['', Validators.required],
+      issuerName: ['', [
+        Validators.required,
+        Validators.maxLength(255),
+        Validators.pattern(/^(?! )(?!.* {3})[\p{L}a-zA-Z0-9\-_ <>&]+$/)
+      ]
+      ],
+      evidenceRequired: ['', Validators.required],
+      enableTaskEvidence: [''],
+      attachLogo: this.fb.array([]),
+      attachSign: this.fb.array([])
+    });
+  }
+
+  onCertificateTypeChange(value: string): void {
+    this.certificateTypeSelected = !!value;
+  }
+
+  attachLogos(){
+    const dialogRef = this.dialog.open(DialogPopupComponent, {
+      disableClose: true,
+      data : {
+        header: "ATTACH_LOGO",
+        subheader: "ATTACH_LOGOS_DETAIlS",
+        subcontent: this.filespecifications,
+        attachButton: "ATTACH"
+      }
+    })
+    dialogRef.afterClosed().subscribe(result => {
+      if (result && result.fileName) {
+        this.attachLogo.push({ name: result.fileName });
+      } 
+    })
+  }
+
+  attachSignature(){
+    const dialogRef = this.dialog.open(DialogPopupComponent, {
+      disableClose: true,
+      data : {
+        header: "ATTACH_SIGNATURE",
+        subheader: "ADD_SIGNATURE_DETALS",
+        inputfields: [
+          { label: "SIGNATURE_NAME", value: "", type: "text" },
+          { label: "SIGNATURE_DESIGNATION", value: "" , type: "text" },
+        ],
+        subcontent: this.filespecifications,
+        attachButton: "ATTACH"
+      }
+    })
+    dialogRef.afterClosed().subscribe(result => {
+      if (result && result.fileName) {
+        this.attachSign.push({ name: result.fileName });
+      } 
+    })
+  }
+
+  removeLogo(index: number) {
+    this.attachLogo.splice(index, 1);
+  }
+
+  removeSignature(index: number) {
+    this.attachSign.splice(index, 1);
+  }
+
+  get attachedLogosData(): FormArray {
```

Comment:
keep as one function and pass FormControl as parameter and return the the data as FormArray